### PR TITLE
Cleanup cycling and configuration of page swappers

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Build page cache factory only page cache is actually created.
Move configure to PageSwapper and remove ConfigurablePageSwapperFactory.

### CODE REVIEW

**Comment by MishaDemianenko** [comment]
@chrisvest can you please take a look? WDYT?

File: community/io/src/main/java/org/neo4j/io/pagecache/PageSwapperFactory.java [hunk 1]
```diff
@@ -33,17 +34,27 @@
...
+
+    /**
+     * Configure swapper with filesystem to use.
+     * @param fs file system to use in page swappers
      */
-    void setFileSystemAbstraction( FileSystemAbstraction fs );
+    default void configure( FileSystemAbstraction fs )
+    {
+        this.configure( fs, Configuration.EMPTY );
+    }
```
Comment (chrisvest) [inline community/io/src/main/java/org/neo4j/io/pagecache/PageSwapperFactory.java]: I think we can get away with only having one `configure` method; the one that takes both parameters.
I'd also like to call it `configureAndInitialise`, to highlight that it has important life-cycle semantics. Or alternatively, we can call it `open`, to be symmetric with the `close` method further down.

**Review by chrisvest — COMMENTED** [review]
Though it is annoying to add a dependency on `neo4j-graphdb-api`, it does actually look like the best solution, since breaking up graphdb-api would be even more uncomfortable.

Just one comment about the `configure` method.

**Comment by MishaDemianenko** [comment]
@chrisvest updated; i have even try to brake that dependency since it was strange in the beginning to have it, but braking and extracting configuration from kernel is quite thing to do :)

**Review by chrisvest — APPROVED** [review]

**Pull request merged**